scripts/ci/test_plan.py

```diff
@@ -4,23 +4,133 @@
 
 # A script to generate twister options based on modified files.
 
-import re, os
+import re
+import os
 import argparse
 import glob
-import yaml
 import json
 import fnmatch
 import subprocess
 import csv
+from pathlib import Path
 import logging
+import concurrent.futures
+import yaml
+
 from git import Repo
 
+
+
 if "ZEPHYR_BASE" not in os.environ:
     exit("$ZEPHYR_BASE environment variable undefined.")
 
 repository_path = os.environ['ZEPHYR_BASE']
 logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
 
+def _procee_maintainer_file(node, zephyr_base_dir, target_files):
+    ''' procee_maintainer_file
+
+        :param node: items to run in multi-process
+        :type node: Hash
+
+        :param zephyr_base_dir: zephyr base dir
+        :type zephyr_base_dir: string
+
+        :param target_files: target files
+        :type target_files: array
+
+        :return Array of matched application path, otherwise empty array
+    '''
+    matched = False
+    test_apps_path = []
+
+    if "files" in node:
+        _test_apps_path = []
+        for item in node["files"]:
+            logging.info(item)
+            if re.match('^tests', item):
+                _test_apps_path.append(item)
+                continue
+            if re.match('^samples', item):
+                _test_apps_path.append(item)
+                continue
+            if os.path.isfile(os.path.join(zephyr_base_dir, item)):
+                __f = os.path.basename(item)
+                if __f in target_files:
+                    matched = True
+            else:
+                if '*' in item:
+                    _files = Path(zephyr_base_dir).rglob(item)
+                    _files = list(set(_files))
+                    __f = [os.path.basename(i) for i in _files]
+                else:
+                    _files = Path(os.path.join(zephyr_base_dir, item)).rglob("*")
+                    _files = list(set(_files))
+                    __f = [os.path.basename(i) for i in _files]
```

Comment:
Can you refactor this part, to reduce repetitiveness?

Reply:
ok